pkg/providers/aws/cluster_network_provider.go

```diff
@@ -26,77 +38,620 @@ import (
+				logger.Infof("found bundled vpc subnet %s in cluster vpc %s", *subnet.SubnetId, *subnet.VpcId)
+			}
+		}
+	}
+	logger.Infof("found %d bundled vpc subnets in cluster vpc", len(validBundledVPCSubnets))
+	return len(validBundledVPCSubnets) == 0, nil
+}
+
+//reconcileStandaloneVPCSubnets returns an array list of private subnets associated with a vpc or an error
+//
+//each standalone vpc cidr block is split in half, to create two private subnets.
+//these subnets are located in different az's
+//the az is determined by the cro strategy, either provided by override config map or provided by the infrastructure CR
+func (n *NetworkProvider) reconcileStandaloneVPCSubnets(ctx context.Context, logger *logrus.Entry, vpc *ec2.Vpc, clusterID, organizationTag string) ([]*ec2.Subnet, error) {
+	logger.Info("gathering all private subnets in cluster vpc")
+
+	// build our subnets, so we know if the vpc /26 then we /27
+	if *vpc.CidrBlock == "" {
+		return nil, errorUtil.New("standalone vpc cidr block can't be empty")
+	}
+
+	// AWS stores it's CIDR block as a string, convert it
+	_, awsCIDR, err := net.ParseCIDR(*vpc.CidrBlock)
+	if err != nil {
+		return nil, errorUtil.Wrapf(err, "failed to parse vpc cidr block %s", *vpc.CidrBlock)
+	}
+	// Get the cluster VPC mask size
+	// e.g. If the cluster VPC CIDR block is 10.0.0.0/8, the size is 8 (8 bits)
+	maskSize, _ := awsCIDR.Mask.Size()
+
+	// If the VPC CIDR mask size is greater or equal to the size that CRO requires
+	// - If equal, CRO will not be able to subdivide the VPC CIDR into sub-networks
+	// - If greater, there will be fewer host addresses available in the sub-networks than CRO needs
+	// Note: The larger the mask size, the less hosts the network can support
+	if maskSize >= defaultSubnetMask {
+		return nil, errorUtil.New(fmt.Sprintf("vpc cidr block %s cannot contain generated subnet mask /%d", *vpc.CidrBlock, defaultSubnetMask))
+	}
+
+	// Split vpc cidr mask by increasing mask by 1
+	halfMaskStr := fmt.Sprintf("%s/%d", awsCIDR.IP.String(), maskSize+1)
+	_, halfMaskCidr, err := net.ParseCIDR(halfMaskStr)
+	if err != nil {
+		return nil, errorUtil.Wrapf(err, "failed to parse half mask cidr block %s", halfMaskStr)
+	}
+
+	// Generate 2 valid sub-networks that can be used in the cluster VPC CIDR range
+	validSubnets := generateAvailableSubnets(awsCIDR, halfMaskCidr)
+	if len(validSubnets) != defaultNumberOfExpectedSubnets {
+		return nil, errorUtil.New(fmt.Sprintf("expected at least two subnet ranges, found %s", validSubnets))
+	}
+
+	// get a list of valid availability zones
+	var validAzs []*ec2.AvailabilityZone
+	azs, err := n.Ec2Api.DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{})
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "error getting availability zones")
+	}
+	// we only require two az's for cro networking
+	for index, az := range azs.AvailabilityZones {
+		validAzs = append(validAzs, az)
+		if index == 1 {
+			break
+		}
+	}
```

Comment:
I think we should sort `azs.AvailabilityZones` here before doing this. It'll ensure the first 2 are always used.